plugins/org.eclipse.n4js.runner/src/org/eclipse/n4js/runner/RuntimeEnvironmentsHelper.java

```diff
@@ -145,11 +146,9 @@ private void getEnvironemntWithAncestorsRecursive(IN4JSProject project, Set<IN4J
 	 * @return list of transitive dependencies of type runtime library, no duplicates
 	 */
 	private List<IN4JSProject> collectRequiredRuntimeLibraries(IN4JSProject project) {
-		if (new SourceContainerAwareDependencyTraverser(project).getResult().hasCycle()) {
-			throw new DependencyCycleDetectedException(project);
-		}
 		Set<IN4JSProject> depsRuntimeLibraries = new HashSet<>();
-		recursiveDependencyCollector(project, depsRuntimeLibraries, p -> isRuntimeLibrary(p));
+		recursiveDependencyCollector(project, depsRuntimeLibraries, p -> isRuntimeLibrary(p),
+				new RecursionGuard<>((new IN4JSProjectEquivalence())));
```

Comment:
Maybe I don't get it, but it seems the old code used the dependency traverser class only to check for a cycle and if the isn't a cylce, it implemented its own dependency collection. This seems redundant (and odd). Don't we have a utility for this? (maybe the `SourceContainerAwareDependencyTraverser` itself)

Reply:
As far as I am familiar with our `DependencyTraverser` and `SourceContainerAwareDependencyTraverser`, the sole purpose of these traversers is to detect dependency cycles (e.g. JavaDoc of `DependencyTraverser` say "TRUE as switch result means: found a cycle and should stop.").